### PR TITLE
Add validation support for SPV_NV_shader_invocation_reorder.

### PR DESCRIPTION
**Pull request by alelenv**
SPV spec : https://github.com/KhronosGroup/SPIRV-Registry/pull/175
GLSL spec : https://github.com/KhronosGroup/GLSL/pull/196

### CODE REVIEW

**Comment by alelenv** [comment]
cc : @dgkoch

@johnkslang or @dneto0 to review
Task list issue contains all related collateral

File: source/val/validate_annotation.cpp [hunk 1]
```diff
@@ -200,7 +200,8 @@ spv_result_t ValidateDecorationTarget(ValidationState_t& _, SpvDecoration dec,
             sc != SpvStorageClassHitAttributeKHR &&
             sc != SpvStorageClassCallableDataKHR &&
             sc != SpvStorageClassIncomingCallableDataKHR &&
-            sc != SpvStorageClassShaderRecordBufferKHR) {
+            sc != SpvStorageClassShaderRecordBufferKHR &&
+            sc != SpvStorageClassHitObjectAttributeNV) {
           return _.diag(SPV_ERROR_INVALID_ID, target)
                  << _.VkErrorID(6672) << _.SpvDecorationString(dec)
```
Comment (spencer-lunarg) [inline source/val/validate_annotation.cpp:206]: There should be a spec update for ` VUID-StandaloneSpirv-Location-06672` to add `HitObjectAttributeNV` storage class

Comment (dgkoch) [reply source/val/validate_annotation.cpp:206]: See https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/5538

**Comment by alan-baker** [comment]
This PR needs rebased.

**Comment by alelenv** [comment]
@alan-baker : Rebased, please re-review

**Comment by alelenv** [comment]
Ping @alan-baker , can you take a look
Thanks

**Review by alan-baker — COMMENTED** [review]
Looks basically good, just need to fix the bots.

**Comment by alelenv** [comment]
@alan-baker : I have fixed the build bot failures. Can you retrigger?

**Comment by alelenv** [comment]
Had missed a clang format, everything is now green except the ndk-build which complains about undefined references

@alan-baker : Can you assist? I am not clear why it would break only on android/ndk build

**Comment by alan-baker** [comment]
You might have forgotten to update other build systems. It looks like only cmakelists are updated. The android.mk needs updated I expect. Bazel is probably glob'ing all the files already.

**Review by alan-baker — APPROVED** [review]

**Pull request merged**